automation/hotswap.sh

```diff
@@ -1,26 +1,33 @@
 #!/usr/bin/env bash
 
-#Updates the Leo jar of a fiab to reflect current local code.
-#Run using "./automation/hotswap.sh fiab-your-fiab-name" from inside the leo directory
+# Updates the Leo jar of a FIAB to reflect current local code.
+# Run using "./automation/hotswap.sh <your FIAB name>" at the root of the leonardo repo clone
 
 if [ -z "$1" ]
     then
-        echo "No arguments supplied. Please provide fiab name as an argument."
+        echo "No arguments supplied. Please provide FIAB name as an argument."
         exit 1
 fi
 
 FIAB=$1
 
+printf "Generating the Leo jar...\n\n"
 # Example: /Users/qi/workspace/leonardo/http/target/scala-2.12/http-assembly-0.1-437ee4a9-SNAPSHOT.jar
-FULL_LEO_JAR=$(sbt -Dsbt.log.noformat=true "project http" assembly | tail -3 | head -2 | grep -o '/Users[^ ]*')
-SHORT_LEO_JAR=$(basename $FULL_LEO_JAR)
-NEW_JAR_NAME=$(echo ${SHORT_LEO_JAR}|sed 's/http\-/leonardo\-/g' ) # rename the jar to leonardo-assembly-0.1-437ee4a9-SNAPSHOT.jar so that fiab start script will pick up the right jar file
+LEO_JAR_PATH=$(sbt -Dsbt.log.noformat=true "project http" assembly | tail -3 | head -2 | grep -o '/Users[^ ]*')
+LEO_JAR_NAME=$(basename $LEO_JAR_PATH)
+# Rename the jar to leonardo-assembly-0.1-437ee4a9-SNAPSHOT.jar so the fiab-start Jenkins job picks up the right jar file
+NEW_LEO_JAR_NAME=$(echo ${LEO_JAR_NAME}|sed 's/http\-/leonardo\-/g' )
 
-gcloud compute scp ${FULL_LEO_JAR} ${FIAB}:/tmp --zone=us-central1-a --project broad-dsde-dev
+printf "Copying ${LEO_JAR_PATH} to /tmp on FIAB '${FIAB}'...\n\n"
+gcloud compute scp ${LEO_JAR_PATH} ${FIAB}:/tmp --zone=us-central1-a --project broad-dsde-dev
 
+printf "Removing the old Leo jars on the FIAB...\n\n"
+# For some reason including the rm command to be executed on the FIAB within the EOSSH block below doesn't work
+gcloud compute ssh --project broad-dsde-dev --zone us-central1-a ${FIAB} -- 'sudo docker exec -it firecloud_leonardo-app_1 sh -c "rm -f /leonardo/*jar"'
+
+printf "Copying the Leo jar to the right location on the FIAB, and restarting the Leo app and proxy...\n\n"
 gcloud compute ssh --project broad-dsde-dev --zone us-central1-a ${FIAB} << EOSSH
```

Comment:
The line below didn't work and had to be moved out of `heredoc` for some reason  🤷‍♂